[DependencyInjection][Framework] `kernel.locale_aware` (#16639)
symfony/symfony-docs: reference/dic_tags.rst

`kernel.event_listener`_                  Listen to different events/hooks in Symfony
`kernel.event_subscriber`_                To subscribe to a set of different events/hooks in Symfony
`kernel.fragment_renderer`_               Add new HTTP content rendering strategies

> `kernel.locale_aware`_                    To interact with the locale

Reviewer: ```suggestion
`kernel.locale_aware`_                    To access and use the current locale
```
?

:class:`Symfony\\Component\\HttpKernel\\Fragment\\FragmentRendererInterface`,
register it as a service, then tag it with ``kernel.fragment_renderer``.

kernel.locale_aware
-------------------

> .. versionadded:: 4.3

Reviewer 2: You need to target v4.4 and here swap to 4.4 :)

Reviewer 3: The merge branch is 4.4, but it's still OK to say that this was introduced in 4.3.

Reviewer 2: @javiereguiluz oki I thought it display only the oldest
also watch out I still see 4.3 in the destination branch

<img width="1214" alt="Capture d’écran 2022-10-26 à 12 08 41" src="https://user-images.githubusercontent.com/1358361/197999315-1ebcf572-acc4-4035-93cf-711c493d2775.png">

    The ``kernel.locale_aware`` tag was introduced in Symfony 4.3.

> **Purpose**: To interact with the locale

Reviewer 2: Perhaps add link:ref to what is a locale in symfony request/response context?

Reviewer: I would remove this line

Reviewer 3: It doesn't add much, but all tags include a "purpose" line, so let's keep it.

Setting and retrieving the locale can be done via the configuration or

> thanks to container parameters or even listeners / route parameters.

Reviewer: ```suggestion
using container parameters, even listeners, route parameters or the current request.
```
?

Thanks to the ``Translation`` contract, the locale can be set via services.

To register your own locale aware service, first create a service that implements
the :class:`Symfony\\Contracts\\Translation\\LocaleAwareInterface` interface::

    // src/Locale/MyCustomLocaleHandler.php
    namespace App\Locale;

    use Symfony\Contracts\Translation\LocaleAwareInterface;

    class MyCustomLocaleHandler implements LocaleAwareInterface
    {
        public function setLocale($locale)
        {

>             // ... do some sort of operations to set the locale

Reviewer: This setter should not be called as it used to pass the locale to the context of the class when built or when using the new switcher.
I would remove this comment and hard code a proper assignation instead:
```php
$this->locale = $locale;
```

        }

        public function getLocale()
        {

>             return 'en';

Reviewer: Following the comment above:
```suggestion
            return $this->locale;
```
